src/lib/classes/lsp3-universal-profile.ts

```diff
@@ -60,16 +67,51 @@ export class LSP3UniversalProfile {
     console.log(version);
   }
 
+  /**
+   * Uploads the LSP3Profile to the desired endpoint. This can be an `https` URL either pointing to
+   * a public, centralized storage endpoint or an IPFS Node / Cluster
+   *
+   * @param {ProfileDataBeforeUpload} profileData
+   * @return {*}  {(Promise<AddResult | string>)}
+   * @memberof LSP3UniversalProfile
+   */
+  static async uploadProfileData(
+    profileData: ProfileDataBeforeUpload,
+    uploadOptions?: ProfileUploadOptions
+  ): Promise<{ profile: LSP3ProfileJSON; url: string }> {
+    uploadOptions = uploadOptions || defaultUploadOptions;
+
+    const profileImage = profileData.profileImage
+      ? await imageUpload(profileData.profileImage, uploadOptions)
+      : null;
+
+    const backgroundImage = profileData.backgroundImage
+      ? await imageUpload(profileData.backgroundImage, uploadOptions)
+      : null;
+
+    const profile = {
+      LSP3Profile: {
+        ...profileData,
+        profileImage,
+        backgroundImage,
+      },
+    };
+
+    // TODO: allow simple http upload too
+    const url = await ipfsUpload(JSON.stringify(profile), uploadOptions.ipfsClientOptions);
```

Comment:
I would suggest to call this the same as in line 29 of uploader

```
const uploadResponse = 
```

Because `url` I would expect it to be a string.

Reply:
Good catch 👍